src/components/MainDrawer/NavigationDrawerButton.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+
+import type { Navigation } from '../../types';
+
+import * as React from 'react';
+import InjectedComponent from '../../common/InjectedComponent';
+import DrawerButton from './DrawerButton';
+import { withNavigation } from 'react-navigation';
+
+type InjectedProps = {|
+  navigation: Navigation,
+|};
+
+type Props = {
+  icon: { name: string, type?: string },
```

Comment:
Would it make sense to make `type` an enum? It might be over the top since we will only use it when adding icons.

Reply:
ok, I'll grab the types from there: https://github.com/react-native-training/react-native-elements/blob/master/src/helpers/getIconType.js

Reply:
I don't think it's a big deal but do what you think is best.